### DISCUSSION TITLE
The 3D model is red or not displayed at all

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1539] [Q&A] [answered]
![orc](https://github.com/axmolengine/axmol/assets/120378517/91887b07-b422-457b-b3d3-4771270c99cb)

In tests, all 3D models are displayed correctly, but in my own project "orc.c3b" without a texture, and "boss1.obj" not displayed at all. Perhaps I missed something in the sample code from the tests, but I double-checked and didn’t notice anything like that.

my update method:
```
case ExampleGameState::init:
{
    _gameState = ExampleGameState::update;

    auto s = Director::getInstance()->getWinSize();

    auto enemy = ax::MeshRenderer::create("orc.c3b");
         enemy->setPosition3D(Vec3(0, 0, 0));
         enemy->setCameraMask(2);
         addChild(enemy);

    auto boss = ax::MeshRenderer::create("boss.obj");
         boss->setPosition3D(Vec3(20.0f, 0, 0));
         boss->setTexture("boss.png");
         boss->setCameraMask(2);
         addChild(boss);

    auto light = ax::PointLight::create(Vec3(0, 0, 20.0f), Color3B(150.0f, 50.0f, 250.0f), 100.0f);
         addChild(light);

    auto camera = Camera::createPerspective(90, s.width / s.height, 1, 1000);
         camera->setCameraFlag(CameraFlag::USER1);
         camera->setPosition3D(Vec3(10.0f, 0.0f, 130.0f));
         addChild(camera);
         camera->lookAt(boss->getPosition3D());

    break;
}
```

**Comment by rh101** [comment, marked as answer]
What is inside the Contents directory of your project? If the `orc.c3b` is not showing the texture, then it may be missing from your project. The file is named `monguger.tga`, and you can find it in the same folder where you copied `orc.c3b` from (`tests\cpp-tests\Content\MeshRendererTest`).

**Reply by 10100010010001001010** [reply]
thanks, it works, I did not know that this was a texture file. c3b, c3t - what are these extensions? I previously tried to find a similar extension, but didn’t find anything relevant

And as i understand for my own 3d models i should use fbx-conv from cocos, right ?

**Reply by rh101** [reply]
> thanks, it works, I did not know that this was a texture file. c3b, c3t - what are these extensions? I previously tried to find a similar extension, but didn’t find anything relevant

First result using web search:
![image](https://github.com/axmolengine/axmol/assets/8603230/0d2aef38-a172-4be9-a00b-cdeffa52f5cb)

> And as i understand for my own 3d models i should use fbx-conv from cocos, right ?

I'm not sure if you *need* to use the c3t/c3b format, as the engine does seem to support the OBJ format.

**Reply by 10100010010001001010** [reply]
thanks for reply, i saw this info about that formats, but it too superficial. I would like to read about the file format in more detail, but unfortunately the creators of cocos did not make any documentation for the format

**Reply by rh101** [reply]
Open the `.c3t` file in a text editor, since it is in plain text format. It should give you some insight regarding the format.

**Comment by rh101** [comment]
The converter tool (modified version of LibGDX fbx-conv) can be found here: https://github.com/cocos2d-x/fbx-conv
